node/messages/src/helpers/codec.rs

```diff
@@ -21,6 +21,9 @@ use ::bytes::{BufMut, BytesMut};
 use core::marker::PhantomData;
 use tokio_util::codec::{Decoder, Encoder, LengthDelimitedCodec};
 
+/// The maximum size of a message that can be transmitted during the handshake.
+const MAXIMUM_HANDSHAKE_MESSAGE_SIZE: usize = 1024 * 1024; // 1 MiB
```

Comment:
How is this being used? I only see it used as the Default for the MessageCodec

Reply:
Its purpose is to limit the size of handshake messages (after which it gets raised to `MAXIMUM_MESSAGE_SIZE` via `MessageCodec::update_max_message_len`) - otherwise any new peer could send us some other large `Message` (e.g. a big `Block`) as one of the handshake messages and we'd be wasting time deserializing it before deciding that we should drop it. It's an anti-DoS measure.